src/components/cc-logs-instances/cc-logs-instances.js

```diff
@@ -0,0 +1,567 @@
+import '../cc-input-text/cc-input-text.js';
+import '../cc-datetime-relative/cc-datetime-relative.js';
+import '../cc-icon/cc-icon.js';
+import '../cc-loader/cc-loader.js';
+import '../cc-toggle/cc-toggle.js';
+import '../cc-notice/cc-notice.js';
+import { css, html, LitElement } from 'lit';
+import {
+  iconRemixCheckboxCircleFill as iconDeploymentOk,
+  iconRemixCheckLine as iconInstanceReady,
+  iconRemixCloseCircleFill as iconDeploymentFailed,
+  iconRemixDeleteBin_2Line as iconHeaderDeleted,
+  iconRemixCloseLine as iconInstanceDeleted,
+  iconRemixPlayCircleLine as iconHeaderRunning,
+  iconRemixPlayLine as iconInstanceRunning,
+  iconRemixSpaceShipLine as iconHeaderDeploying,
+  iconRemixStopCircleFill as iconDeploymentCancelled,
+  iconRemixStopCircleLine as iconHeaderStopping,
+  iconRemixStopLine as iconInstanceStopping,
+  iconRemixToolsLine as iconBuildInstance,
+  iconRemixGitCommitLine as iconGitCommit,
+} from '../../assets/cc-remix.icons.js';
+import { dispatchCustomEvent } from '../../lib/events.js';
+import { i18n } from '../../lib/i18n.js';
+import { groupBy } from '../../lib/utils.js';
+
+/**
+ * @param {Instance} i1
+ * @param {Instance} i2
+ * @return {number}
+ */
+const INSTANCE_SORT_ORDER = (i1, i2) => {
+  const number = i1.index - i2.index;
+  if (number !== 0) {
+    return number;
+  }
+  return i1.creationDate.getTime() - i2.creationDate.getTime();
+};
+
+/**
+ * @typedef {import('./cc-logs-instances.types.d.ts').LogsInstancesState} LogsInstancesState
+ */
+
+/**
+ * Allows users to select some instances.
+ *
+ * * When using `live` mode, instances are grouped by logical state.
+ * * When using `cold` mode, instances are grouped by deployment.
+ *
+ * @cssdisplay flex
+ *
+ * @event {CustomEvent} cc-logs-instances:selectionChange - Fires whenever the instances selection changes
+ */
+export class CcLogsInstances extends LitElement {
+
+  static get properties () {
+    return {
+      state: { type: Object },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {LogsInstancesState} */
```

Comment:
nitpick: on other components we provide a description next to the type. For the state we tend to do something like `Sets the state of...` or `State of the ...`, could we do that here too?